### ISSUE TITLE
Provide a possibility to bypass two factor auth

### ISSUE THREAD

**Issue by sgiehl** [sgiehl/piwik-plugin-GoogleAuthenticator#12]
In the case that someone looses his possibility to generate tokens (e.g. lost his phone) there should be a possibility to bypass the two factor auth in order to be still able to login to piwik.

The possibility provided by other services is to generate a list of one time tokens.

A possibility suggest by a user would be to bypass two factor auth for specific IPs (which won't work for dynamic IPs)

**Comment by Draky50110** [comment]
Subscribing to one time token, as it is mainly used by Google or other company to bypass in case of...

**Comment by mannp** [comment]
I would also like to know the answer to this please and am not fully clear of your response Draky, could you expand a little please?

**Comment by Draky50110** [comment]
Other big websites like Google let us use bypass codes.

**Comment by mannp** [comment]
Agreed they do, so do nextcloud and others, but this plugin doesn't support it does it.

**Comment by sgiehl** [comment]
I would love to be able to add support for this, but I'm currently not able handle this. Pull Requests are welcome :)

**Comment by mannp** [comment]
Just trying to understand any potential lockout scenarios. Are we able to disable the plugin locally and re-login to recover the account?

If we edit the configs that is.

Thanks

**Comment by sgiehl** [comment]
Sure. You can disable the plugin to disable it for all accounts. Or you can remove the settings for a specific account from the database to disable it for a specific account:
piwik_option table; option_name is GoogleAuthenticator.username

**Comment by mannp** [comment]
Perfect thanks and does disabling it leave the original username and pass in tact, merely removing the additional 2fa step.

So we can login as usual?

**Comment by sgiehl** [comment]
exactly. it only disables the 2Fa step.

**Comment by mannp** [comment]
That works for me, thanks :) and thanks for your work.

**Comment by jookk** [comment]
how to do it exactly, please... 
I am locked out, looked to tables, but cant find it
![pwk](https://user-images.githubusercontent.com/11217085/32989984-130056aa-cd21-11e7-8adf-db235e4a4758.png)

**Comment by sgiehl** [comment]
@jookk Try `DELETE FROM piwik_option WHERE option_name = 'GoogleAuthenticator.username' LIMIT 1;`, where you need to replace `username` with your own users login. Afterwards you need to set up 2FA for that user again

**Comment by jookk** [comment]
so, I tried and... 
![pwk](https://user-images.githubusercontent.com/11217085/33025833-4b0f99fe-ce0f-11e7-844d-c4c80bf1a1e2.png)

**Comment by sgiehl** [comment]
then it shouldn't be active for your user. what exactly is your problem when logging in?

**Comment by jookk** [comment]
Yes, it was active and I used it successfuly. Recently changed phone and TfA is gone. 
BUT, i found ./console plugins list and deactivate... 
So, I simply deactivated Google Auth plugin and logged in. :)
